pkg/operator/controller/canary/daemonset_test.go

```diff
@@ -0,0 +1,60 @@
+package canary
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func TestDesiredCanaryDaemonSet(t *testing.T) {
+	canaryImage := "openshift/hello-openshift:latest"
+
+	daemonset := desiredCanaryDaemonSet(canaryImage)
+
+	expectedDaemonSetName := types.NamespacedName{
+		Namespace: "openshift-ingress-canary",
+		Name:      "ingress-canary",
+	}
+
+	if !cmp.Equal(daemonset.Name, expectedDaemonSetName.Name) {
+		t.Errorf("Expected daemonset name to be %s, but got %s", expectedDaemonSetName.Name, daemonset.Name)
+	}
+
+	if !cmp.Equal(daemonset.Namespace, expectedDaemonSetName.Namespace) {
+		t.Errorf("Expected daemonset namespace to be %s, but got %s", expectedDaemonSetName.Namespace, daemonset.Namespace)
+	}
+
+	expectedLabels := map[string]string{
+		"ingress.openshift.io/canary": "canary_controller",
+	}
+
+	if !cmp.Equal(daemonset.Labels, expectedLabels) {
+		t.Errorf("Expected daemonset labels to be %q, but got %q", expectedLabels, daemonset.Labels)
+	}
+
+	labelSelector := &metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			"app": "ingress-canary",
```

Comment:
Should we follow the [k8s label recommendations](https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/) instead of "app: ingress-canary"?

Reply:
fixed. Thanks!